tidy3d/components/utils.py

```diff
@@ -0,0 +1,106 @@
+"""Utilities shared by multiple components go here."""
+
+from __future__ import annotations
+
+from typing import Any, Optional
+
+import shapely
+
+from tidy3d.components.types import Shapely
+from tidy3d.log import log
+
+
+def pop_axis_and_swap(
```

Comment:
would rather this live in the same place as `pop_axis_and_swap`

Reply:
You mean `pop_axis()`?
That was my plan originally.  `pop_axis_and_swap()` was originally defined in `components/geometry/base.py` (next to `pop_axis()`).

However later (in PR2544) I moved `pop_axis_and_swap()` out of that folder so that I could use it inside the plot functions in the `plugins/microwave/` folder.  (Example [here](https://github.com/flexcompute/tidy3d/pull/2544/files#diff-c29589882a85bdccb040c1250253b6a4295d2492fb000f02757f63583f891fdd).)